frontend/packages/dev-console/src/components/pipelines/pipeline-form/MultipleResourceKeySelector.tsx

```diff
@@ -0,0 +1,106 @@
+import { RootState } from '@console/internal/redux';
+import { getActiveNamespace } from '@console/internal/reducers/ui';
+import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { ResourceDropdown, getFieldId, useFormikValidationFix } from '@console/shared';
+import { FormGroup } from '@patternfly/react-core';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import MultipleKeySelector from './MultipleKeySelector';
+
+interface MultipleResourceKeySelectorProps {
+  label: string;
+  resourceModel: K8sKind;
+  required?: boolean;
+  resourceNameField: string;
+  resourceKeysField: string;
+}
+
+interface StateProps {
+  namespace: string;
+}
+
+const MultipleResourceKeySelector: React.FC<StateProps & MultipleResourceKeySelectorProps> = ({
+  label,
+  namespace,
+  resourceModel,
+  required,
+  resourceNameField,
+  resourceKeysField,
+}) => {
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const [field, { touched, error }] = useField(resourceNameField);
+  const isValid = !(touched && error);
+  const fieldId = getFieldId(resourceNameField, 'res-dropdown');
+  const [keys, setKeys] = React.useState({});
+
+  useFormikValidationFix(field.value);
+
+  const resource: WatchK8sResource = React.useMemo(
+    () => ({
+      kind: resourceModel.kind,
+      isList: true,
+      namespace,
+      optional: true,
+    }),
+    [namespace, resourceModel.kind],
+  );
+
+  const [resources, loaded, loadError] = useK8sWatchResource(resource);
+
+  const autocompleteFilter = (strText, item): boolean => fuzzy(strText, item?.props?.name);
+
+  const generateKeys = (resourceName: string) => {
+    const selectedResource: K8sResourceKind = _.find(resources, (res) => {
+      return _.get(res, 'metadata.name') === resourceName;
+    });
+    const keyMap = selectedResource?.data;
+    const itemKeys = Object.keys(keyMap).reduce((acc, key) => ({ ...acc, [key]: key }), {});
+    setKeys(itemKeys);
+  };
+
+  return (
+    <FormGroup
+      fieldId={fieldId}
+      label={label}
+      isValid={isValid}
+      className="odc-multiple-key-selector"
+      isRequired={required}
+    >
+      <ResourceDropdown
+        resources={[
+          { kind: resourceModel.kind, loaded, loadError, data: resources as K8sResourceKind[] },
+        ]}
+        loaded={loaded}
+        loadError={loadError}
+        dataSelector={['metadata', 'name']}
+        selectedKey={field.value}
+        placeholder={`Select a ${resourceModel.label}`}
+        autocompleteFilter={autocompleteFilter}
+        dropDownClassName={cx({ 'dropdown--full-width': true })}
+        onChange={(value: string) => {
+          setFieldValue(resourceNameField, value);
+          setFieldTouched(resourceNameField, true);
+          generateKeys(value);
+        }}
+        showBadge
+      />
+      <MultipleKeySelector name={resourceKeysField} keys={keys} />
```

Comment:
We need to disable or hide the sub section in `MultipleResourceKeySelector` until the ResourceDropdown has a value.

![image](https://user-images.githubusercontent.com/8126518/79339896-92045200-7ef7-11ea-97da-a2b728ec79e3.png)

Simplest solution I can see is this:
```suggestion
      {field.value && <MultipleKeySelector name={resourceKeysField} keys={keys} />}
```
But it doesn't clean up some of the form quirks. Like switching between two ResourceDropdown fields don't clear the `MultipleKeySelector` fields. 

If at all possible (hopefully it's easy), reset that form on `ResourceDropdown`'s `onChange`

Reply:
There is also a design issue we need to work around. I've logged a bug to deal with it after this is merged
https://issues.redhat.com/browse/ODC-3571

Let's try to keep our changes as small as possible to get everything merged.

Reply:
Thanks for logging the bug.

Reply:
Along with with this I will also add what I suggested here https://coreos.slack.com/archives/CHG0KRB7G/p1586957157163200